### PR TITLE
feat: improved docs view and refactored divider

### PR DESCRIPTION
**Pull request by noahwaldner**
- improved MDX Parsing
- cleaned Divider component
- improved component navigation
	- added sorting
- fixed icon generation issues
- added script to watch MDX files

### CODE REVIEW

File: docs/src/components/EditableComponent/CodeSnippet.tsx [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+            await navigator.clipboard.writeText(code);
+            successCallback();
+        } catch (error) {
+            throw new Error('Failed to copy code to clipboard');
+        }
+    };
+    if (code === undefined || code === null || code === '') {
+        return null;
+    }
+    return (
```
Comment (SamuelAlev) [inline docs/src/components/EditableComponent/CodeSnippet.tsx:28]:
```suggestion
    };

    if (code === undefined || code === null || code === '') {
        return null;
    }

    return (
```

File: docs/src/components/EditableComponent/CopyCodeButton.tsx
Comment (SamuelAlev) [inline docs/src/components/EditableComponent/CopyCodeButton.tsx:1]: Please make the code a bit more airy by adding line breaks (every files)

File: docs/src/components/EditableComponent/CopyCodeButton.tsx [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+                </span>
+                <div className="tw-bg-[#011627] tw-z-10">
+                    <IconClipboard size={20} />
+                </div>
+            </div>
+            <div
+                className={`tw-transition-all tw-duration-500 tw-flex tw-overflow-hidden tw-absolute tw-right-0 tw-bottom-0 ${!copiedSuccessfully ? 'tw-translate-y-full tw-opacity-0' : 'tw-opacity-1'}`}
+            >
+                <span className={'tw-pr-2 tw-whitespace-nowrap'}>Copied Successfully</span>
+                <div className="tw-bg-[#011627] tw-z-10">
```
Comment (SamuelAlev) [inline docs/src/components/EditableComponent/CopyCodeButton.tsx:43]: please use tokens

File: docs/src/components/EditableComponent/CopyCodeButton.tsx [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+                    Copy to clipboard
+                </span>
+                <div className="tw-bg-[#011627] tw-z-10">
+                    <IconClipboard size={20} />
+                </div>
+            </div>
+            <div
+                className={`tw-transition-all tw-duration-500 tw-flex tw-overflow-hidden tw-absolute tw-right-0 tw-bottom-0 ${!copiedSuccessfully ? 'tw-translate-y-full tw-opacity-0' : 'tw-opacity-1'}`}
+            >
+                <span className={'tw-pr-2 tw-whitespace-nowrap'}>Copied Successfully</span>
```
Comment (SamuelAlev) [inline docs/src/components/EditableComponent/CopyCodeButton.tsx:42]:
```suggestion
                <span className="tw-pr-2 tw-whitespace-nowrap">Copied Successfully</span>
```

File: docs/src/components/EditableComponent/CopyCodeButton.tsx [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+        >
+            <div
+                className={`tw-transition-all tw-duration-500 tw-flex tw-overflow-hidden tw-absolute tw-right-0 tw-bottom-0 ${copiedSuccessfully ? '-tw-translate-y-full tw-opacity-0' : 'tw-opacity-1'}`}
+            >
+                <span
+                    className={`tw-pr-2 tw-transition-all tw-duration-500 tw-whitespace-nowrap  ${isHovered ? 'tw-opacity-100' : 'tw-translate-x-full tw-opacity-0'}`}
+                >
+                    Copy to clipboard
+                </span>
+                <div className="tw-bg-[#011627] tw-z-10">
```
Comment (SamuelAlev) [inline docs/src/components/EditableComponent/CopyCodeButton.tsx:35]: Please use tokens

File: docs/src/components/EditableComponent/CopyCodeButton.tsx [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+            onMouseEnter={() => setIsHovered(true)}
+            onMouseLeave={() => setIsHovered(false)}
+            onClick={handleButtonClick}
+            className="tw-absolute tw-rounded-br tw-mx-2 tw-bottom-2 tw-right-0 tw-text-sm tw-text-text-weak "
+        >
+            <div
+                className={`tw-transition-all tw-duration-500 tw-flex tw-overflow-hidden tw-absolute tw-right-0 tw-bottom-0 ${copiedSuccessfully ? '-tw-translate-y-full tw-opacity-0' : 'tw-opacity-1'}`}
+            >
+                <span
+                    className={`tw-pr-2 tw-transition-all tw-duration-500 tw-whitespace-nowrap  ${isHovered ? 'tw-opacity-100' : 'tw-translate-x-full tw-opacity-0'}`}
```
Comment (SamuelAlev) [inline docs/src/components/EditableComponent/CopyCodeButton.tsx:31]:
```suggestion
                    className={`tw-pr-2 tw-transition-all tw-duration-500 tw-whitespace-nowrap ${isHovered ? 'tw-opacity-100' : 'tw-translate-x-full tw-opacity-0'}`}
```

File: docs/src/components/EditableComponent/CopyCodeButton.tsx [hunk 6]
```diff
@@ -0,0 +1,49 @@
...
+                setCopiedSuccessfully(false);
+            }, 3000);
+        });
+    };
+    return (
+        <button
+            onMouseEnter={() => setIsHovered(true)}
+            onMouseLeave={() => setIsHovered(false)}
+            onClick={handleButtonClick}
+            className="tw-absolute tw-rounded-br tw-mx-2 tw-bottom-2 tw-right-0 tw-text-sm tw-text-text-weak "
```
Comment (SamuelAlev) [inline docs/src/components/EditableComponent/CopyCodeButton.tsx:25]:
```suggestion
            className="tw-absolute tw-rounded-br tw-mx-2 tw-bottom-2 tw-right-0 tw-text-sm tw-text-text-weak"
```

File: packages/components/src/components/Divider/Divider.mdx [hunk 7]
```diff
@@ -0,0 +1,54 @@
+---
+title: Divider
+type: Component
+exportedName: Divider, DividerHeight
+---
+
+## API
```
Comment (SamuelAlev) [inline packages/components/src/components/Divider/Divider.mdx:7]: What about:
- Live example with dummy data (the "simple")
- Anatomy (show the different components in a simple way so we know what is available)
Example: 
```tsx
<Dialog>
  <DialogHeader></DialogHeader>
  <DialogContent></DialogContent>
  <DialogFooter></DialogFooter>
</Dialog>
```
- API
- Examples

File: packages/components/src/components/Divider/Divider.mdx [hunk 8]
```diff
@@ -0,0 +1,54 @@
...
+
+| Name     | Type                                          | Description              |
```
Comment (SamuelAlev) [inline packages/components/src/components/Divider/Divider.mdx:9]: Replace `description` with `default`?

**Review by SamuelAlev — CHANGES_REQUESTED** [review]

File: package.json [hunk 9]
```diff
@@ -14,7 +24,9 @@
     "devDependencies": {
         "@changesets/changelog-github": "^0.5.0",
         "@changesets/cli": "^2.27.1",
-        "glob": "^10.3.10"
+        "concurrently": "^8.2.2",
+        "glob": "^10.3.10",
+        "npm-watch": "^0.11.0"
```
Comment (SamuelAlev) [inline package.json:29]: Could you prefer `chokidar`?

File: packages/components/src/components/Divider/Divider.mdx (hunk 7 above)
Comment (noahwaldner) [reply packages/components/src/components/Divider/Divider.mdx:7]: This is a Divider not a Dialog ^^

There aren't really any different components

Comment (SamuelAlev) [reply packages/components/src/components/Divider/Divider.mdx:7]: I know, I just propose having a generic and default structure

Comment (noahwaldner) [reply packages/components/src/components/Divider/Divider.mdx:7]: agree with this then!

File: docs/src/components/EditableComponent/CopyCodeButton.tsx
Comment (noahwaldner) [reply docs/src/components/EditableComponent/CopyCodeButton.tsx:1]: added

File: packages/components/src/components/Divider/Divider.mdx (hunk 8 above)
Comment (noahwaldner) [reply packages/components/src/components/Divider/Divider.mdx:9]: changed

File: package.json (hunk 9 above)
Comment (noahwaldner) [reply package.json:29]: switched to cookidar, i don't have strong feelings for or against any of them

Comment (SamuelAlev) [reply package.json]: `npm-watch` uses `nodemon` which uses `chokidar` 😄

**Review by SamuelAlev — APPROVED** [review]
Approving as issue can be fixed in separate PR

**Pull request merged**